packages/cli/src/commands/join.ts

```diff
@@ -231,6 +232,9 @@ packageVersion: string
           const pathOperation = paths[path][operation];
           joinedDef.paths[path][operation] = pathOperation;
           potentialConflicts.paths[path][operation] = [...(potentialConflicts.paths[path][operation] || []), entrypoint];
+          if (operation === '$ref') {
+            break; // do not proceed with reference, it's a string path
```

Comment:
Don't we have to resolve the `$ref`?

Reply:
right now we don't resolve it, but I'll add this feature within this PR